ArduSub/mode_guided.cpp

```diff
@@ -227,6 +263,24 @@ void ModeGuided::guided_set_velocity(const Vector3f& velocity)
     position_control->set_vel_desired_cms(velocity);
 }
 
+// guided_set_velocity - sets guided mode's target velocity
+void ModeGuided::guided_set_velocity(const Vector3f& velocity, bool use_yaw, float yaw_cd, bool use_yaw_rate, float yaw_rate_cds, bool relative_yaw)
+{
+   // check we are in velocity control mode
+    if (sub.guided_mode != Guided_Velocity) {
+        guided_vel_control_start();
+    }
+
+    // set yaw state
+    guided_set_yaw_state(use_yaw, yaw_cd, use_yaw_rate, yaw_rate_cds, relative_yaw);
+
+    update_time_ms = AP_HAL::millis();
+
+    // set position controller velocity target
+    position_control->set_vel_desired_cms(velocity);
+
+}
+
 // set guided mode posvel target
 bool ModeGuided::guided_set_destination_posvel(const Vector3f& destination, const Vector3f& velocity)
```

Comment:
This variation of `ModeGuided::guided_set_destination_posvel` does not call `ModeGuided::guided_set_yaw_state`, so AFAICT it does not initialize variables that are referenced in `ModeGuided::guided_posvel_control_run`. @Williangalvani